Add socket_zmq to zyre.api file and minor documentation updates

Dysl3xik opened the pull request:
Followup for PR #659

sphaero commented on api/zyre.api:
```diff
@@ -284,6 +284,12 @@
+	<method name = "socket_zmq">
+        Return underlying ZMQ socket for talking to the Zyre node, 
+        for polling with libzmq (base ZMQ library)
+        <return type = "void *" />
```
Can you try to change the type to "anything" or perhaps "sockish"? See: https://github.com/zeromq/zproject#api-types

bluca commented on api/zyre.api:
```diff
@@ -284,6 +284,12 @@
         <return type = "zsock" />
     </method>
+	
+	<method name = "socket_zmq">
```
Add state = "draft"

Dysl3xik commented:
Thanks for the input guys I am still learning how this project is setup! Its a little more complex than the normal C/C++ ones I have experience with.

brettviren commented:
Since CZMQ is required by Zyre the function `zsock_resolve()` is available to all Zyre apps.  Maybe what's helpful here is to add a reference to this function and why it may be useful in the Zyre documentation?  Maybe a sentence at the end of README in the section "Hints to Contributors" is enough?  Or a new section "Hints to Users" could be started?

Dysl3xik commented:
Why I added this:
I added this function here as I thought it made the library a little more user friendly as the function you noted is in czmq not zyre, and its difficult to keep track of whats where and whats available. 

Additionally comments in the API like the existing socket function state: "Return socket for talking to the Zyre node, for polling" although this is factually correct it glosses over the fact that it return a czmq socket and not a zmq socket.

I originally fell in a trap where I was trying to poll using zmq (the other library I am using for direct ZMQ functionality uses libzmq) and feeding in the socket provided by the existing socket function here in zyre. Everything will build fine in that case it just gives exceptions at run time which are also unclear.

Anyways you may not find this PR useful, but as a newcomer coming to ZeroMQ and trying to mesh different libs together this was certainly an issue for me. In my own code I now annotate everything as either czmq_thing or zmq_thing to help with this issue as I have users filling out messages that can either be passed to Zyre or ZMQ directly and I have to hide all this behind the scenes.

sphaero commented:
If it's considered not useful we'll remove it again. Are you able to regenerate using zproject? As all bindings need to be updated as well.

Dysl3xik commented:
I am unfamiliar with running that sorry and I don't have a machine setup for python or other language bindings to actually test building any of that stuff. I only have windows.

bluca commented:
I've pushed a commit with the regen. It should be possible to run from Windows, but I don't use it so not entirely sure.

zedalaye commented:
> It should be possible to run from Windows, but I don't use it so not entirely sure.

No,as far as I experienced, GSL do not compile on Windows. I'm used to run GSL from a Linux box or from WSL (Windows Subsystem for Linux)
